### PR TITLE
Add bilateral texture filter

### PR DESCRIPTION
**Pull request by jhlee525**
_I sent a pull request few months ago (#550), but I upload again because of my git commit issues._

Bilateral texture filter [1], along with its test code, is added.

Bilateral texture filter is one of its kind for texture smoothing and edge preserving. This method blurs high-frequency texture area while preserving structural edges including object borders. It is done with a simple modification of the original bilateral filter. This method can be applicable to many other researches including feature extraction, depth image noise reduction and detail manipulation. This method was accepted in Siggraph 2014 as an oral presentation.

[1] Cho et al., Bilateral Texture Filtering, in proceedings of SIGGRAPH 2014

### CODE REVIEW

**Comment by mshabunin** [comment]
@alalek , it is same as PR #550 that you've reviewed already.

File: modules/ximgproc/src/bilateral_texture_filter.cpp [hunk 1]
```diff
@@ -0,0 +1,357 @@
...
+          }
+        }
+      }
+    }
+  }
+
+  void joint_bilateral_filter(const Mat& img, const Mat& G, Mat& r_img, int fr2, double sigma_avg)
+	{
+		Mat p_G;
+		copyMakeBorder(G, p_G, fr2, fr2, fr2, fr2, BORDER_REFLECT);
```
Comment (alalek) [inline modules/ximgproc/src/bilateral_texture_filter.cpp]: @jhlee525 Please use spaces instead of tabs.

**Comment by jhlee525** [comment]
I changed all tab to space in bilateral_texture_filter.cpp file. Thank you for feedback

**Comment by alalek** [comment]
Thank you!
:+1:

**Pull request merged**